### ISSUE TITLE
 The resolved config file doesn't contain a resolve configuration

### ISSUE THREAD

**Issue by angelosembrano** [RyanCCollins/scalable-react-boilerplate#43]
I haven't configured anything yet, so it is the first instance  of my ```npm run start``` after ```npm run setup```.

This shows:
```
Error: D:/Projects/DIR/scalable-react-boilerplate/webpack.config.babel.js: The resolved config file doesn't contain a resolve configuration
    at PluginPass.CallExpression (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-plugin-webpack-alias\build\index.js:48:27)
    at newFn (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-traverse\lib\visitors.js:276:21)
    at NodePath._call (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-traverse\lib\path\context.js:76:18)
    at NodePath.call (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-traverse\lib\path\context.js:48:17)
    at NodePath.visit (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-traverse\lib\path\context.js:105:12)
    at TraversalContext.visitQueue (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-traverse\lib\context.js:150:16)
    at TraversalContext.visitSingle (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-traverse\lib\context.js:108:19)
    at TraversalContext.visit (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-traverse\lib\context.js:192:19)
    at Function.traverse.node (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-traverse\lib\index.js:114:17)
    at NodePath.visit (D:\Projects\DIR\scalable-react-boilerplate\node_modules\babel-traverse\lib\path\context.js:115:19)
```
Why is that? I am developing on a Windows 10 environment.

**Comment by Gothfrid** [comment]
guess i have same error:
`WebpackOptionsValidationError: Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration has an unknown property 'postcss'. These properties are valid:
   For typos: please correct them.
   For loader options: webpack 2 no longer allows custom properties in configuration.
 - configuration.module has an unknown property 'preLoaders'. 
`
Ubuntu 16.04 \ node v6.9.1 \ npm  v3.10.8

**Comment by RyanCCollins** [comment]
Hey! Thanks for reporting this.  I just updated the package.json to indicate that the webpack version needs to be exactly as listed.  If you remove your node_modules and reinstall them after pulling the latest version, this should be working.  Sorry for that!

The webpack version needs to read:
```
    "webpack": "2.1.0-beta.19"
```
in the package.json

**Comment by angelosembrano** [comment]
Still gething this @RyanCCollins.

`npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.15: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN babel-plugin-webpack-loaders@0.7.1 requires a peer of webpack@>=1.12.9 <3.0.0 but none was installed.
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\SmartwaveDev\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v6.4.0
npm ERR! npm  v3.10.9
npm ERR! path D:\Projects\DIR\scalable-react-boilerplate\node_modules\.staging\diff-ce2b4922\package.json.642187904
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall rename

npm ERR! Error: EPERM: operation not permitted, rename 'D:\Projects\DIR\scalable-react-boilerplate\node_modules\.staging\diff-ce2b4922\package.json.642187904' -> 'D:\Projects\DIR\scalable-react-boilerplate\node_modules\.staging\diff-ce2b4922\package.json'
npm ERR!     at Error (native)
npm ERR!  { Error: EPERM: operation not permitted, rename 'D:\Projects\DIR\scalable-react-boilerplate\node_modules\.staging\diff-ce2b4922\package.json.642187904' -> 'D:\Projects\DIR\scalable-react-boilerplate\node_modules\.staging\diff-ce2b4922\package.json'
npm ERR!     at Error (native)
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'rename',
npm ERR!   path: 'D:\\Projects\\DIR\\scalable-react-boilerplate\\node_modules\\.staging\\diff-ce2b4922\\package.json.642187904',
npm ERR!   dest: 'D:\\Projects\\DIR\\scalable-react-boilerplate\\node_modules\\.staging\\diff-ce2b4922\\package.json',
npm ERR!   parent: 'scalable-react-boilerplate' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\Projects\DIR\scalable-react-boilerplate\npm-debug.log

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\SmartwaveDev\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "setup"
npm ERR! node v6.4.0
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE
npm ERR! scalable-react-boilerplate@1.1.0 setup: `npm install`
npm ERR! Exit status 4294963248
npm ERR!
npm ERR! Failed at the scalable-react-boilerplate@1.1.0 setup script 'npm install'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the scalable-react-boilerplate package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm install
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs scalable-react-boilerplate
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls scalable-react-boilerplate
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\Projects\DIR\scalable-react-boilerplate\npm-debug.log`

**Comment by angelosembrano** [comment]
Update:

Went past by the error above  by running ```npm run setup``` through an Administrator command prompt.

But still, it shows the same error about resolve configuration.

Skimming through the list of npm packages, I noticed this is the only one having issues:
```+-- UNMET PEER DEPENDENCY webpack@2.1.0-beta.19```

Why is that?

**Comment by RyanCCollins** [comment]
It looks like you do not have the proper version of webpack installed.  

I would recommend starting from scratch with the installation.  Also, we've found that one of the biggest sources of bugs comes from when users are not on the same version of Node and NPM.

The versions listed in the package.json are

```
  "engines": {
    "node": "5.2.0",
    "npm": "3.10.7"
  },
```

I usually use NVM to install specific versions of Node.  You can see a [Gist here](https://gist.github.com/RyanCCollins/1a5686ff9dd51b72eb2d4dc70aa6c1f4).  When you install node 5.2.0, it should also install the right version of npm.

So to summarize, what I recommend doing is installing Node version 5.2.0 with NVM and then start with a fresh install of the boilerplate `git clone https://github.com/RyanCCollins/scalable-react-boilerplate` then `npm run setup`.  Also, if you have Yarn installed, you can skip npm run setup and just do `yarn`.

Good luck! Let me know how it goes.

**Comment by RyanCCollins** [comment]
I am going to close this. Please message me if you still need help @angelosembrano
